news/2023-12-21-improving-supply-chain-security.md

```diff
@@ -0,0 +1,86 @@
+
+This PKI would be managed by the Rust Project and (more specifically) its infrastructure team, with support from the Rust Foundation.
+
+Part of defining the PKI includes defining how certificates will be managed, delegated, rotated, and revoked if necessary; getting to consensus on these details will be achieved through the initial PKI RFC.
+
+### Releases
+
+The canonical way Rust users get releases today is via `rustup`. At present, this uses simple HTTPS downloads from Rust Project infrastructure, which uses the system certificate store to validate that `rustup` is connecting to the right servers. While certificate pinning could be employed to mitigate issues with compromised system certificate stores, that doesn't help address problems for those who need to use mirrors.
+
+We intend to create a delegated certificate within the new Rust PKI that can sign release artifacts, and then to work with the `rustup` team to extend `rustup` to be able to verify that artifacts are signed by the release team using the delegated certificate.
+
+### Crates
+
+Similar concerns apply to crates — Rustaceans need to be able to download crates and know that they're getting the crate files that were published to crates.io without modification.
+
+We have a head start here: the crate index (both sparse and Git) includes SHA-256 checksums. This means that we can sign the index while getting the security guarantees that we need, without having to immediately sign crate files themselves. We also have the benefit of [prior work on index signing](https://github.com/rust-lang/rfcs/pull/2474), which we are grateful to be able to learn from.
+
+Our plan is to use another delegated certificate to sign each index entry along with the index as a whole. Doing so allows `cargo` to validate that crates haven't been tampered with after publishing, and also allows the state of the entire index to be checked, making it harder for a malicious mirror to not update a vulnerable crate. This will be done in a layered way that allows for expansion, allowing the chain of trust to be extended as [future work](#future-possibilities) is implemented.
+
+You may have noticed that this doesn't address crate signing by authors; for more detail on that, see the [future possibilities section](#future-possibilities).
+
+## When
+
+We're starting on this work immediately: we'll be opening an RFC for the foundational PKI portion of this plan Real Soon Now™, and following it with RFCs for the release and crates components thereafter.
+
+## Who
+
+The Rust Foundation’s security initiative team are leading this effort in conjunction with the Rust Project, consulting with other domain experts and stakeholders as needed. We're using the [`#tbd-signing` Zulip stream](https://rust-lang.zulipchat.com/#narrow/stream/417663-tbd-signing) to co-ordinate, and welcome feedback and suggestions there.
```

Comment:
"The Rust Foundation's Security Initiative team is leading this effort [...]"

Reply:
Changed to "Technology Team" to match the above; otherwise left alone to be true to my state's weirdly mangled version of English grammar.